ecs-agent/credentials/instancecreds/instancecreds_linux.go

```diff
@@ -28,15 +28,27 @@ import (
 // GetCredentials returns the instance credentials chain. This is the default chain
 // credentials plus the "rotating shared credentials provider", so credentials will
 // be checked in this order:
+//
 //  1. Env vars (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY).
+//
 //  2. Shared credentials file (https://docs.aws.amazon.com/ses/latest/DeveloperGuide/create-shared-credentials-file.html) (file at ~/.aws/credentials containing access key id and secret access key).
+//
 //  3. EC2 role credentials. This is an IAM role that the user specifies when they launch their EC2 container instance (ie ecsInstanceRole (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/instance_IAM_role.html)).
+//
 //  4. Rotating shared credentials file located at /rotatingcreds/credentials
+//
+//     In the case of external instance `RotatingSharedCredentialsProvider` provider should be prioritized over
+//     `EC2RoleProvider` one as SSM credentials are going to be ignored in case agent is deployed to EC2
+//     with instance role which doesn't have permissions needed to run ECS Agent
 func GetCredentials(isExternal bool) *credentials.Credentials {
```

Comment:
Looks like this change makes `GetCredentials` implementation for Linux identical to that of Windows. I think we should merge the two into a single `instancecreds/instancecreds.go` file.

Reply:
In the meanwhile I will reproduce this issue on my end. Once the PR is ready I can verify that the change works by testing it in the same environment.

Reply:
I was able to reproduce the issue on my end.

Reply:
Since merging the Windows implementation with Linux will increase the scope of testing, I am okay with making this change for Linux alone in this PR and do the merge as future work (maybe leave a TODO here). 

I noticed that the Windows implementation gives precedence to rotating shared creds over shared credentials file even (since it slots in rotating creds provider at second position in the slice instead of third) which does not match the documentation of the function. 

https://github.com/aws/amazon-ecs-agent/blob/b702281f0f6aa2fba5614050769aa337d1456853/ecs-agent/credentials/instancecreds/instancecreds_windows.go#L28-L72